File: app/src/main/java/br/com/ufpe/cin/myfootprints/LocationListener.java
package br.com.ufpe.cin.myfootprints;

import android.content.Context;
import android.location.Location;
import android.os.Bundle;
import android.util.Log;

import java.util.Date;

public class LocationListener implements android.location.LocationListener {

    private static final String TAG = "LOCATION_LISTENER";
    private Location previousLocation;
    private LocationUpdateDAO dbInstance;

    public LocationListener(String provider, Context ctx) {
        previousLocation = new Location(provider);
        dbInstance = LocationUpdateDAO.getInstance(ctx);
    }

    public long currentTimestampSeconds() {
        Date date = new Date();
        return date.getTime();
    }

    @Override
    public void onLocationChanged(Location location) {
        previousLocation.set(location);

        if (location.getAccuracy() <= 150f) {
            LocationUpdate update = new LocationUpdate(
                    previousLocation.getLatitude(),
                    previousLocation.getLongitude(),
                    (int)(currentTimestampSeconds()/1000)
            );

            dbInstance.insertLocationUpdate(update);
        }
    }

    @Override
    public void onProviderDisabled(String provider) {}

    @Override
    public void onProviderEnabled(String provider) {}

    @Override
    public void onStatusChanged(String provider, int status, Bundle extras) {}
}

File: app/src/main/java/br/com/ufpe/cin/myfootprints/LocationListener.kt
package br.com.ufpe.cin.myfootprints

import android.content.Context
import android.location.Location
import android.os.Bundle
import android.util.Log

import java.util.Date

class LocationListener(provider: String, ctx: Context) : android.location.LocationListener {
    private val previousLocation: Location
    private val dbInstance: LocationUpdateDAO?

    init {
        previousLocation = Location(provider)
        dbInstance = LocationUpdateDAO.getInstance(ctx)
    }

    fun currentTimestampSeconds(): Long {
        val date = Date()
        return date.time
    }

    override fun onLocationChanged(location: Location) {
        previousLocation.set(location)

        if (location.accuracy <= 150f) {
            val update = LocationUpdate(
                    previousLocation.latitude,
                    previousLocation.longitude,
                    (currentTimestampSeconds() / 1000).toInt()
            )

            dbInstance!!.insertLocationUpdate(update)
        }
    }

    override fun onProviderDisabled(provider: String) {}

    override fun onProviderEnabled(provider: String) {}

    override fun onStatusChanged(provider: String, status: Int, extras: Bundle) {}

    companion object {

        private val TAG = "LOCATION_LISTENER"
    }
}
